state/machine.go

```diff
@@ -347,6 +347,30 @@ func (m *Machine) IsManual() (bool, error) {
 	return false, nil
 }
 
+// IsIsolated returns true if the machine networking should be ran by fan
+// TODO this returns true for all containers on models with FAN configured,
+// it should use value inherited from '--isolated' cmdline flag
+func (m *Machine) IsIsolated() bool {
+	if !m.IsContainer() {
+		return false
+	}
+	cfg, err := m.st.ModelConfig()
+	if err != nil {
+		logger.Errorf("Error getting model config %v", err)
+		return false
+	}
+	if len(cfg.FanConfig()) > 0 {
+		return true
+	} else {
+		return false
+	}
+}
+
+func (m *Machine) ModelSubnets() ([]*Subnet, error) {
```

Comment:
This seems clumbsy. I don't think a machine should return model subnets.

Reply:
Removed.